### PR TITLE
Bump libthrift

### PR DESCRIPTION
**Pull request by Sam-Kramer**
## General
**Before this PR**:
Libthrift is old 

**After this PR**:

==COMMIT_MSG==
Bumps libthrift to 0.17.0
==COMMIT_MSG==

**Priority**:
P1

**Concerns / possible downsides (what feedback would you like?)**:
Doesn't break the world upon release? 

**Is documentation needed?**:
No

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
Can introduce API breaks if clients manually set libthrift to a lower version 

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:
Yes

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No

**Does this PR need a schema migration?**
No

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
N/A

**What was existing testing like? What have you done to improve it?**:
N/A

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:
N/A

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:
N/A

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:
Logs/metrics indicating that a service is unable to talk to C*. 

**Has the safety of all log arguments been decided correctly?**:
N/A

**Will this change significantly affect our spending on metrics or logs?**:
No

**How would I tell that this PR does not work in production? (monitors, etc.)**:
Logs/metrics indicating that a service is unable to talk to C*. C* p99 latency regressions. 

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
Revert this PR and https://github.com/palantir/cassandra/pull/381

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:
@mdaudali for your awareness! 

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:
Potentially -- libthrift bump may introduce performance regressions, but none have yet to be observed. 

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:
No

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:
No

## Development Process
**Where should we start reviewing?**:
Now!

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: versions.props [hunk 1]
```diff
@@ -95,11 +95,11 @@ ch.qos.logback:* = 1.2.11
 # Database drivers can have unexpected breaking changes and should be upgraded with caution
 com.datastax.cassandra:cassandra-driver-core = 3.8.0
 com.oracle.database.jdbc:ojdbc11 = 21.5.0.0
-com.palantir.cassandra:* = 2.2.18-1.101.0
+com.palantir.cassandra:* = 2.2.18-1.108.0
```
Comment (jeremyk-91) [inline versions.props:98]: Yes: this version has the new libthrift too.

**Review by jeremyk-91 — COMMENTED** [review]
Do we need to update the product dependencies on `sls-cassandra` for this one?

File: versions.props (hunk 1 above)
Comment (Sam-Kramer) [reply versions.props:98]: Cassandra-thrift is required for the bump, but sls-cassandra itself does not need libthrift 0.17.0 for this to be compatible, and visa versa, sls-cassandra on libthrift 0.17.0 does not require clients to be on this version

Comment (jeremyk-91) [reply versions.props:98]: Ah okay, it's the interface with the `cassandra-thrift` library as opposed to a wireformat thing - got it

**Review by jeremyk-91 — APPROVED** [review]
👍

**Pull request merged**